rules/Php85/Rector/StmtsAwareInterface/NestedToPipeOperatorRector.php

```diff
@@ -0,0 +1,315 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Php85\Rector\StmtsAwareInterface;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\Cast\Array_;
+use PhpParser\Node\Expr\Cast\String_;
+use PhpParser\Node\Expr\ConstFetch;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Scalar\Float_;
+use PhpParser\Node\Scalar\Int_;
+use PhpParser\Node\Stmt;
+use PhpParser\Node\Stmt\Expression;
+use PhpParser\Node\VariadicPlaceholder;
+use Rector\Contract\PhpParser\Node\StmtsAwareInterface;
+use Rector\Rector\AbstractRector;
+use Rector\ValueObject\PhpVersionFeature;
+use Rector\VersionBonding\Contract\MinPhpVersionInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see https://wiki.php.net/rfc/pipe-operator-v3
+ * @see \Rector\Tests\Php85\Rector\StmtsAwareInterface\NestedToPipeOperatorRector\NestedToPipeOperatorRectorTest
+ */
+final class NestedToPipeOperatorRector extends AbstractRector implements MinPhpVersionInterface
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Transform nested function calls and sequential assignments to pipe operator syntax',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+$value = "hello world";
+$result1 = function3($value);
+$result2 = function2($result1);
+$result = function1($result2);
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+$value = "hello world";
+
+$result = $value
+    |> function3(...)
+    |> function2(...)
+    |> function1(...);
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [StmtsAwareInterface::class];
+    }
+
+    public function provideMinPhpVersion(): int
+    {
+        return PhpVersionFeature::PIPE_OPERATOER;
+    }
+
+    public function refactor(Node $node): ?Node
+    {
+        if (! $node instanceof StmtsAwareInterface || $node->stmts === null) {
```

Comment:
The Node already `StmtsAwareInterface` per defined in `getNodeTypes()`:

```suggestion
    /**
      * @param StmtsAwareInterface $node
      */
    public function refactor(Node $node): ?Node
    {
        if ($node->stmts === null) {
```